src/commands/remoteDebug/startRemoteSsh.ts

```diff
@@ -0,0 +1,110 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { SiteConfigResource, User } from 'azure-arm-website/lib/models';
+import * as portfinder from 'portfinder';
+import * as vscode from 'vscode';
+import { SiteClient, TunnelProxy } from 'vscode-azureappservice';
+import { callWithTelemetryAndErrorHandling, IActionContext } from 'vscode-azureextensionui';
+import { SiteTreeItem } from '../../explorer/SiteTreeItem';
+import { WebAppTreeItem } from '../../explorer/WebAppTreeItem';
+import { ext } from '../../extensionVariables';
+import { delay } from '../../utils/delay';
+import * as remoteDebug from './remoteDebugCommon';
+
+const remoteSsh: Map<string, boolean> = new Map();
+
+export async function startRemoteSsh(node?: SiteTreeItem): Promise<void> {
+    if (!node) {
+        node = <SiteTreeItem>await ext.tree.showTreeItemPicker(WebAppTreeItem.contextValue);
+    }
+    if (remoteSsh.get(node.root.client.fullName)) {
+        throw new Error(`Azure Remote SSH is currently starting or already started for "${node.root.client.fullName}".`);
+    }
+
+    remoteSsh.set(node.root.client.fullName, true);
+    try {
+        await startRemoteSshInternal(node);
+    } catch (error) {
+        remoteSsh.set(node.root.client.fullName, false);
+        throw error;
+    }
+}
+
+async function startRemoteSshInternal(node: SiteTreeItem): Promise<void> {
+    const siteClient: SiteClient = node.root.client;
+    const siteConfig: SiteConfigResource = await siteClient.getSiteConfig();
+    const oldSetting: boolean = <boolean>siteConfig.remoteDebuggingEnabled;
+    // should always be an unbound port
+    const portNumber: number = await portfinder.getPortPromise();
+
+    await vscode.window.withProgress({ location: vscode.ProgressLocation.Notification }, async (progress: vscode.Progress<{}>): Promise<void> => {
+        if (!siteConfig.linuxFxVersion) {
+            throw new Error('Azure Remote SSH is only supported for Linux web apps.');
+        }
+
+        remoteDebug.reportMessage('Checking app settings...', progress);
+
+        // remote debugging has to be disabled in order to tunnel to the 2222 port
+        await remoteDebug.setRemoteDebug(false, undefined /*skips confirmation*/, undefined, siteClient, siteConfig);
+
+        remoteDebug.reportMessage('Starting tunnel proxy...', progress);
```

Comment:
`Starting tunnel proxy...` is probably meaningless for most users. Can we get a better message?

Reply:
Sure. Any suggestions?

Something simple like `Starting SSH...`?

Reply:
Yeah that's better